resources/terraform/modules/ec2/variables.tf

```
@@ -45,14 +45,20 @@ variable "vpc_cidr_block" {
   default     = "10.1.0.0/20"
 }
 
-variable "subnet_private_cidr_block" {
-  description = "The cidr block of the private subnet"
-  default     = "10.1.1.0/24"
-}
-
-variable "subnet_public_cidr_block" {
-  description = "The cidr block of the public subnet"
-  default     = "10.1.2.0/24"
+variable "subnets" {
+  description = "Subnets configuration"
+  type = object({
+    private = object({
+      count = number
+    })
+    public = object({
+      count = number
+    })
+  })
+  validation {
+    condition     = var.subnets.private.count > 0 || var.subnets.public.count > 0
```

Does it make sense to have only private subnets without Internet access? What would be the use case? :thinking: 
```suggestion
    condition     = (var.subnets.private.count > 0 && var.subnets.public.count > 0) || var.subnets.public.count > 0
```

---

Discussed with @sk4zuzu that having private subnet implies that you need public one as well.